solidity/contracts/l2/L2WormholeGateway.sol

```diff
@@ -0,0 +1,275 @@
+///            Portal and selects one of the L2 chains they want to bridge from.
+///         2. The holder submits one transaction to L2 that is burning the
+///            token. After the transaction is mined, they wait about 15 minutes
+///            for the L2 block finality.
+///         3. The holder submits one transaction to L1 unlocking their tBTC
+///            tokens from the bridge’s smart contract. After that transaction
+///            is mined, they have their tBTC on L1.
+///
+///         This smart contract is integrated with step 3 of L1->L2 bridging and
+///         step 1 of L2->L1 or L2->L2 bridging. When the user redeems token on
+///         L2, this contract receives the Wormhole tBTC representation and
+///         mints the canonical tBTC in an equal amount. When user sends their
+///         token from L1, this contract burns the canonical tBTC and sends
+///         Wormhole tBTC representation through the bridge in an equal amount.
+/// @dev This contract is supposed to be deployed behind a transparent
+///      upgradeable proxy.
+contract L2WormholeGateway is
+    Initializable,
+    OwnableUpgradeable,
+    ReentrancyGuardUpgradeable
+{
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    /// @notice Reference to the Wormhole Token Bridge contract.
+    IWormholeTokenBridge public bridge;
+
+    /// @notice Wormhole tBTC token representation.
+    IERC20Upgradeable public bridgeToken;
+
+    /// @notice Canonical tBTC token.
+    L2TBTC public tbtc;
+
+    /// @notice Minting limit for this gateway. Useful for early days of testing
+    ///         the system. The gateway can not mint more canonical tBTC than
+    ///         this limit.
+    uint256 public mintingLimit;
+
+    /// @notice The amount of tBTC minted by this contract. tBTC burned by this
+    ///         contract decreases this amount.
+    uint256 public mintedAmount;
+
+    event WormholeTbtcReceived(address receiver, uint256 amount);
+
+    event WormholeTbtcSent(
+        uint256 amount,
+        uint16 recipientChain,
+        bytes32 recipient,
+        uint256 arbiterFee,
+        uint32 nonce
+    );
+
+    event MintingLimitUpdated(uint256 mintingLimit);
+
+    function initialize(
+        IWormholeTokenBridge _bridge,
+        IERC20Upgradeable _bridgeToken,
+        L2TBTC _tbtc
+    ) external initializer {
+        __Ownable_init();
+        __ReentrancyGuard_init();
+
+        require(
+            address(_bridge) != address(0),
+            "Wormhole bridge address must not be 0x0"
+        );
+        require(
+            address(_bridgeToken) != address(0),
+            "Bridge token address must not be 0x0"
+        );
+        require(
+            address(_tbtc) != address(0),
+            "L2TBTC token address must not be 0x0"
+        );
+
+        bridge = _bridge;
+        bridgeToken = _bridgeToken;
+        tbtc = _tbtc;
+        mintingLimit = type(uint256).max;
```

Comment:
Depending on whether you want to mint everything from the get-go or if you want more control over the minting limit, you may want to set this value explicitly to zero (to be clear about the intention) and then require that the owner call `updateMintingLimit` after initialization.